clang/lib/Sema/SemaChecking.cpp

```diff
@@ -7885,6 +7885,18 @@ ExprResult Sema::BuildAtomicExpr(SourceRange CallRange, SourceRange ExprRange,
   if ((IsOpenCL || IsHIP || IsScoped) &&
       Op != AtomicExpr::AO__opencl_atomic_init)
     ++AdjustedNumArgs;
+
+  // Verify if the arguments are of type CompleteType
```

Comment:
When doing so, please see if you can 'extract' this to a function, something like;

`RequireAtomicArgs(Args)` kind of thing.